### PR TITLE
Refactor simplesearcher to make RM3 work on MB corpus and results of simplesearcher consistent with searchCollection

### PR DESCRIPTION
**Pull request by Victor0118**
Add a field for tweet search to make sure it works with RM3. 

See: https://github.com/castorini/Anserini/blob/master/src/main/java/io/anserini/rerank/lib/Rm3Reranker.java#L109

@lintool @Peilin-Yang Could you take a look at this PR?

### CODE REVIEW

File: src/main/java/io/anserini/search/SimpleSearcher.java [hunk 1]
```diff
@@ -82,6 +84,16 @@ public SimpleSearcher(String indexDir) throws IOException {
     setNormalReranker();
```
Comment (lintool) [inline src/main/java/io/anserini/search/SimpleSearcher.java]: this seems to be old code, but can re rename "Default" as opposed to "Normal"?

File: src/main/java/io/anserini/search/SimpleSearcher.java [hunk 2]
```diff
@@ -82,6 +84,16 @@ public SimpleSearcher(String indexDir) throws IOException {
...
   }
 
+  public void setSearchTweets(boolean flag) {
+     this.searchtweets = flag;
+     if (flag) {
+        this.analyzer = new TweetAnalyzer();
+      }
+     else {
```
Comment (lintool) [inline src/main/java/io/anserini/search/SimpleSearcher.java]: else on line above.

```
} else {
```

File: src/main/java/io/anserini/search/SimpleSearcher.java [hunk 3]
```diff
@@ -82,6 +83,10 @@ public SimpleSearcher(String indexDir) throws IOException {
     setNormalReranker();
   }
 
+  public void setSearchTweets(boolean flag) {
```
[the remark this answers is not in the document]
Comment (lintool) [reply src/main/java/io/anserini/search/SimpleSearcher.java]: I think the point here is for this class to be used from Python?

**Comment by Victor0118** [comment]
@lintool @Peilin-Yang The refactor is done!

File: src/main/java/io/anserini/search/SimpleSearcher.java [hunk 4]
```diff
@@ -139,24 +161,50 @@ public void close() throws IOException {
...
     SearchArgs searchArgs = new SearchArgs();
     searchArgs.arbitraryScoreTieBreak = false;
     searchArgs.hits = k;
-    RerankerContext context = new RerankerContext<>(searcher, null, query, null, q, queryTokens, null, searchArgs);
+    searchArgs.searchtweets = searchtweets;
+
+    TopDocs rs = new TopDocs(0, new ScoreDoc[]{}, Float.NaN);
+    RerankerContext context;
+    if (searchtweets) {
+		if (t > 0) {
```
Comment (lintool) [inline src/main/java/io/anserini/search/SimpleSearcher.java]: fix two-space indentation here...

File: src/main/java/io/anserini/search/SimpleSearcher.java [hunk 5]
```diff
@@ -139,24 +161,50 @@ public void close() throws IOException {
...
     ScoredDocuments hits = cascade.run(ScoredDocuments.fromTopDocs(rs, searcher), context);
 
     Result[] results = new Result[hits.ids.length];
     for (int i = 0; i < hits.ids.length; i++) {
       Document doc = hits.documents[i];
-      String docid = doc.getField(LuceneDocumentGenerator.FIELD_ID).stringValue();
-      IndexableField field = doc.getField(LuceneDocumentGenerator.FIELD_RAW);
+      String docid = doc.getField(FIELD_ID).stringValue();
+      IndexableField field = doc.getField(FIELD_RAW);
       String content = field == null ? null : field.stringValue();
```
Comment (lintool) [inline src/main/java/io/anserini/search/SimpleSearcher.java]: Switch back to `LuceneDocumentGenerator.FIELD_RAW` - don't do static imports.

**Comment by lintool** [comment]
Ref: https://github.com/castorini/Anserini/pull/519#issuecomment-447676537
According to @Victor0118 we addressed the issue referenced above ^^^^

**Comment by Victor0118** [comment]
@lintool You comments resolved.

**Pull request merged**